recv.c

```diff
@@ -93,7 +93,7 @@ int MPI_Irecv(void *buf, int count, MPI_Datatype datatype,
       return(MPI_SUCCESS);
     }
 
-  rreq->buf=buf;
+  rreq->buf=(int*)buf;
```

Comment:
Is this type cast correct?

Reply:
Yes, rreq->buf is int* type in the code.

Reply:
Is the right change here to change Req->buf to void * instead?

Reply:
> Is the right change here to change Req->buf to void * instead?

I did not know why buf is defined as int* in the struct Req. Need confirmation before we can change it to void* instead:
```
typedef struct
{
  pListitem listitem;        /* to allow Req to be removed from list */

  int *buf;
  int source;
  int tag;
  int complete;
  int count;
  MPI_Datatype type;  /* needed for test_simple_bindexed2 */
} Req;
```

Reply:
Looks like the user buffer associated with the request to me. I would change it to void *

Reply:
OK, I will update it.